packages/ra-supabase-core/src/authProvider.ts

```diff
@@ -0,0 +1,101 @@
+import { AuthProvider, UserIdentity } from 'ra-core';
+import { SupabaseClient, User } from '@supabase/supabase-js';
+
+export const supabaseAuthProvider = (
+    client: SupabaseClient,
+    { getIdentity }: SupabaseAuthProviderOptions
+): SupabaseAuthProvider => ({
+    async login({ email, password }: { email: string; password: string }) {
+        const { error } = await client.auth.signIn({ email, password });
+
+        if (error) {
+            throw error;
+        }
+        return undefined;
+    },
+    async setPassword({
+        access_token,
+        password,
+    }: {
+        access_token: string;
+        password: string;
+    }) {
+        const { error } = await client.auth.api.updateUser(access_token, {
+            password,
+        });
+
+        if (error) {
+            throw error;
+        }
+        return undefined;
+    },
+    async logout() {
+        const { error } = await client.auth.signOut();
+        if (error) {
+            throw error;
+        }
+    },
+    checkError() {
```

Comment:
the fact that you mix async functions and functions returning a Promise is confusing

Reply:
Why?

Reply:
because when one property is async, the fact that the others aren't makes me think there is a mistake. And also, the code style of each function is different, forcing me to an additional mental effort